tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/PromptValidationTool/DownloadPromptsTool.cs

```diff
@@ -0,0 +1,297 @@
+            string sourceRepoName, 
+            string destinationPath, 
+            List<string> filesToDownload)
+        {
+            var downloadedFiles = new List<string>();
+
+            logger.LogInformation("Starting download of {fileCount} files", filesToDownload.Count);
+
+            foreach (var filePath in filesToDownload)
+            {
+                var fileName = Path.GetFileName(filePath);
+                
+                // Fetch the file content from GitHub
+                logger.LogInformation("Fetching content for {fileName} from {filePath}", fileName, filePath);
+                var contentsResult = await gitHubService.GetContentsAsync(sourceRepoOwner, sourceRepoName, filePath);
+                var fileContent = contentsResult?[0]; // since we expect a single file per path
+
+                if (fileContent?.Content == null)
+                {
+                    var error = $"Failed to fetch content for file {fileName} from {filePath}";
+                    logger.LogError(error);
+                    throw new InvalidOperationException(error);
+                }
+
+                var localFilePath = Path.Combine(destinationPath, fileName);
+                await File.WriteAllTextAsync(localFilePath, fileContent.Content);
+                
+                downloadedFiles.Add(fileName);
+                logger.LogInformation("Downloaded {fileName} from {sourcePath} to {localPath}", fileName, filePath, localFilePath);
+            }
+
+            // Add downloaded files to .gitignore
+            if (downloadedFiles.Count > 0)
+            {
+                await AddToGitignoreAsync(destinationPath, downloadedFiles);
+            }
+
+            logger.LogInformation("Download completed: {downloadedCount}/{totalCount} files successful", 
+                downloadedFiles.Count, filesToDownload.Count);
+
+            return new DownloadResponse
+            {
+                Success = true,
+                DownloadedCount = downloadedFiles.Count,
+                TotalFiles = filesToDownload.Count
+            };
+        }
+
+        /// <summary>
+        /// Adds downloaded files to .gitignore to prevent them from being committed
+        /// </summary>
+        private async Task AddToGitignoreAsync(string destinationPath, List<string> downloadedFiles)
+        {
+            try
+            {
+                // Find git repository root by walking up the directory tree
+                var repoRoot = FindGitRepositoryRoot(destinationPath);
+                if (repoRoot == null)
+                {
+                    logger.LogWarning("Not in a git repository, skipping .gitignore update");
+                    return;
+                }
+
+                // Skip .gitignore update if we're in the azure-rest-api-specs repository
+                var repoName = Path.GetFileName(repoRoot);
+                if (string.Equals(repoName, "azure-rest-api-specs", StringComparison.OrdinalIgnoreCase))
```

Comment:
I think this will become irrelevant if we short circuit the call in the beginning if repo name in the request is same as repo wea redownloading from.